File: api/src/main/java/com/michaelcruz/api/entities/Carro.java
package com.michaelcruz.api.entities;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;
import java.util.Set;

@Data
@Entity
@Table(name = "carro")
public class Carro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idcarro")
    private Long id;

    @Column(name = "nomecarro")
    private String nome;

    @Column(name = "anofabricacaocarro")
    private int anoFabricacao;

    @Column(name = "anomodelocarro")
    private int anoModelo;

    @Column(name = "modelocarro")
    private String modelo;

    @ManyToOne
    @JoinColumn(name = "marca_idmarca")
    private Marca marca;

    @ManyToMany
    @JoinTable(name = "carro_cor",
            joinColumns = @JoinColumn(name = "carro_idcarro"),
            inverseJoinColumns = @JoinColumn(name = "cor_idcor"))
    private List<Cor> cores;

}

File: frontend/src/app/model/Carro.ts
import { Cor } from "./Cor";
import { Marca } from "./Marca";

export class Carro {
  id: number = 0;
  nome: string;
  anoFabricacao: number;
  anoModelo: number;
  modelo: string;
  marca: Marca;
  cores: Cor[];

  constructor(nome: string,
              anoFabricacao: number,
              anoModelo: number,
              modelo: string,
              marca: Marca,
              cores: Cor[]){

    this.nome = nome;
    this.anoFabricacao = anoFabricacao;
    this.anoModelo = anoModelo;
    this.modelo = modelo;
    this.marca = marca;
    this.cores = cores
  }
}
